cypress-io/cypress
# feat: expose projects on global welcome page

Add projects from graphql to GlobalPage. Allows user to double click to select a project to set as the active project.

### Demo

https://user-images.githubusercontent.com/8095629/135011537-3942991c-73fc-49e6-b199-ee75373b80ef.mov

### How to Test
If no projects in your cache first populate it
1. yarn dev
2. navigate to localhost:52159/graphql
3. Enter the mutation with your path to launchpad
```gql
mutation {
  addProject(path: "/Users/cesar/repos/cypress/packages/launchpad") {
    projects {
      id
      title
      projectId
      projectRoot
      isFirstTimeCT
      isFirstTimeE2E
    }
  }
}
```
4. Hit play button
5. Restart dev server

If/when you have data in your cache
1. At root run `yarn dev`
2. GlobalPage appears with list of projects from your cache
3. Double click a project

### PR Tasks
- [ ] Has the original issue or this PR been tagged with a release in ZenHub? 
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?
